File: src/main/java/ru/ioleksiv/telegram/bot/api/model/objects/inline/queryresult/InlineQueryResultVideo.java
package ru.ioleksiv.telegram.bot.api.model.objects.inline.queryresult;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * @see <a href="https://core.telegram.org/bots/api#inlinequeryresultvideo>InlineQueryResultVideo</a>
 */
public class InlineQueryResultVideo extends TitledInlineResult {
    public static final String TYPE = "video";
    /**
     * video_url String A valid URL for the embedded video player or video file
     */
    @JsonProperty("video_url")
    private String videoUrl = null;
    /**
     * mime_type String Mime type of the content of video url, “text/html” or “video/mp4”
     */
    @JsonProperty("mime_type")
    private String mimeType = null;
    /**
     * thumb_url String URL of the thumbnail (jpeg only) for the video
     */
    @JsonProperty("thumb_url")
    private String thumbUrl = null;
    /**
     * video_width Integer Optional. Video width
     */
    @JsonProperty("video_width")
    private Integer videoWidth = null;
    /**
     * video_height Integer Optional. Video height
     */
    @JsonProperty("video_height")
    private Integer videoHeight = null;
    /**
     * video_duration Integer Optional. Video duration in seconds
     */
    @JsonProperty("video_duration")
    private Integer videoDuration = null;
    /**
     * description String Optional. Short description of the result
     */
    @JsonProperty("description")
    private String description = null;

    InlineQueryResultVideo() {
        super(TYPE);
    }

    public String getVideoUrl() {
        return videoUrl;
    }

    public void setVideoUrl(String videoUrl) {
        this.videoUrl = videoUrl;
    }

    public String getMimeType() {
        return mimeType;
    }

    public void setMimeType(String mimeType) {
        this.mimeType = mimeType;
    }

    public String getThumbUrl() {
        return thumbUrl;
    }

    public void setThumbUrl(String thumbUrl) {
        this.thumbUrl = thumbUrl;
    }

    public Integer getVideoWidth() {
        return videoWidth;
    }

    public void setVideoWidth(Integer videoWidth) {
        this.videoWidth = videoWidth;
    }

    public Integer getVideoHeight() {
        return videoHeight;
    }

    public void setVideoHeight(Integer videoHeight) {
        this.videoHeight = videoHeight;
    }

    public Integer getVideoDuration() {
        return videoDuration;
    }

    public void setVideoDuration(Integer videoDuration) {
        this.videoDuration = videoDuration;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/objects/inline/queryresult/InlineQueryResultVideo.kt
package io.github.oleksivio.telegram.bot.api.model.objects.inline.queryresult

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * @see [InlineQueryResultVideo](https://core.telegram.org/bots/api.inlinequeryresultvideo)
 */
data class InlineQueryResultVideo(
        /**
         * video_url String A valid URL for the embedded video player or video file
         */
        @JsonProperty("video_url")
        var videoUrl: String,
        /**
         * mime_type String Mime type of the content of video url, “text/html” or “video/mp4”
         */
        @JsonProperty("mime_type")
        var mimeType: String,
        /**
         * thumb_url String URL of the thumbnail (jpeg only) for the video
         */
        @JsonProperty("thumb_url")
        var thumbUrl: String,
        /**
         * video_width Integer Optional. Video width
         */
        @JsonProperty("video_width")
        var videoWidth: Int? = null,
        /**
         * video_height Integer Optional. Video height
         */
        @JsonProperty("video_height")
        var videoHeight: Int? = null,
        /**
         * video_duration Integer Optional. Video duration in seconds
         */
        @JsonProperty("video_duration")
        var videoDuration: Int? = null,
        /**
         * description String Optional. Short description of the result
         */
        @JsonProperty("description")
        var description: String? = null,

        /**
         * type String Type of the result
         */
        @JsonProperty("type")
         val type: String  = "video"

): TitledInlineResult()
